pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -406,15 +416,6 @@ if [[ ! -z "$specialisation" && ! "$action" = switch && ! "$action" = test ]]; t
     exit 1
 fi
 
-tmpDir=$(mktemp -t -d nixos-rebuild.XXXXXX)
-
-if [[ ${#tmpDir} -ge 60 ]]; then
-    # Very long tmp dirs lead to "too long for Unix domain socket"
-    # SSH ControlPath errors. Especially macOS sets long TMPDIR paths.
-    rmdir "$tmpDir"
-    tmpDir=$(TMPDIR= mktemp -t -d nixos-rebuild.XXXXXX)
-fi
-
 cleanup() {
```

Comment:
This cleanup routine is connected to the tmpdir, and the ssh options. Could you move this too, so that it appears right after the tmpdir construction?
Otherwise the cleanup may not run for the code in between, and it'd be harder to see the connection.